pyproject.toml

```diff
@@ -101,8 +101,7 @@ dependencies = [
 "xformers" = [
   # Core generation dependencies, pinned for reproducible builds.
   "xformers>=0.0.28.post1; sys_platform!='darwin'",
-  # Auxiliary dependencies, pinned only if necessary.
-  "triton; sys_platform=='linux'",
```

Comment:
As far as I can tell, CUDA builds of torch 2.4.1 have a dependency on triton, which has two consequences:
1. triton is installed without us declaring the dependency here.
2. some versions of triton *conflict* with that torch dependency. i.e. torch 2.4.1+cu124 requires 3.0.0 and will conflict with 3.1.0.

If we had an exact dependency on `torch==2.4.1+cu124`, uv's resolver would figure it out which version of triton works, and it would be fine.

However, since the torch version is left ambiguous, it's possible for the resolver to decide that non-CUDA torch==2.4.1 is a better solution since that doesn't conflict with the latest version of xformers. 😖

Reply:
We'll hopefully be bumping torch soon to support the new Nvidia GPUs, so we'll address it then. There's also an additional quirk of requiring a different torch pin for AMD, so the rabbit hole will go even deeper.

Reply:
yeah, and uv has a documentation page about torch specifically: https://docs.astral.sh/uv/guides/integration/pytorch/#configuring-accelerators-with-optional-dependencies

I did make an attempt at implementing that here, but its scope quickly grew far past what I wanted to deal with in this docker-focused PR.